api/envoy/api/v2/BUILD

```diff
@@ -86,6 +86,7 @@ api_proto_library_internal(
         "//envoy/api/v2/core:base",
         "//envoy/api/v2/listener",
         "//envoy/api/v2/listener:udp_listener_config",
+        "//envoy/config/filter/network/http_connection_manager/v2:http_connection_manager",
```

Comment:
I think this causes a circular build loop for proto packages as extensions depend on v2 transitively.

Reply:
Understood.  But it's not obvious how to resolve this problem.  Any suggestions...?

Reply:
We might need to make a decision whether HCM belongs to the core, or whether HTTP API listener belongs to an extension.

Reply:
@markdroth let's resolve the discussion below then thinks about this, if we get rid of the port to HCM map then we don't need it.

Or just use Filter message which is the opaque config supports HCM.

Reply:
I'm fine with waiting until some of the broader issues here are resolved.  However, I don't think the discussion below will actually make this issue go away.  Even if we remove the port map, we will still want an `HttpConnectionManager` field directly in the listener for the HTTP API listener, since the whole point of this new listener type is that it is always hard-coded to a single `HttpConnectionManager`.

Anyway, let's see how things shake out with the broader design discussion below, and then we can come back to this.